### PR TITLE
Fix failure of retrieving git hash from a hg-git enabled repository

### PR DESCRIPTION
**Pull request by mihaipopescu**
It also improves a bit the messaging when there is a failure retrieving the hash from a VCS which can be at times confusing #1137.

This re-integrates 3ef03cd6ac3fe218e77f14336e7c17daf75ba551 back.

### CODE REVIEW

**Comment by CDanU** [comment]
Uff, I just wanted to started reviewing this.

File: scripts/make_version.py [hunk 1]
```diff
@@ -42,20 +42,20 @@ def main(args):
...
         try:
             check_call(['hg', 'gexport'])
-            proc0 = Popen(['hg', '--config', 'defaults.log=', 'log', '-r', '.', '--template', '{gitnode}'], stdout=PIPE, stderr=PIPE)
+            proc0 = Popen(['hg', '--config', 'defaults.log=', 'log', '-r', '.', '--template', '{gitnode}'], stdout=PIPE, stderr=PIPE, cwd=root)
             node_b, error_txt_b = proc0.communicate()
             node = node_b.decode("UTF-8")
             error_txt = "%d: %s" % (proc0.returncode, error_txt_b.decode("UTF-8").strip().lower())
 
-            proc1 = Popen(['git', '--git-dir=.hg/git', 'describe', '--tags', '--always','--dirty', node], stdout=PIPE, stderr=PIPE)
+            proc1 = Popen(['git', '--git-dir=.hg/git', 'describe', '--long', '--tags','--always', node], stdout=PIPE, stderr=PIPE, cwd=root)
```
Comment (CDanU) [inline scripts/make_version.py]: Why are those changed? If I remember correctly I removed the long to prevent a hash string on a tagged commit (the actual release). The -dirty flag should not be removed as it is useful information when we receive bug reports. It could be that the uncommitted _dirty_ changes are causing bugs.

Comment (mihaipopescu) [inline scripts/make_version.py]: It's a special setup as I'm not under git but it's an hg repository with a git remote using hggit extension.

I'll try without --long but -dirty doesn't work in my case. It fails with a strange error (iirc something about comittish, was funny).

Comment (mihaipopescu) [inline scripts/make_version.py]: I think this is the closest I ended up matching the git version but indeed, maybe some things have changed. Sorry not taking this into account.
